lib/rubygems/package/tar_reader.rb

```diff
@@ -117,6 +97,8 @@ def seek(name) # :yields: entry
 
     return yield found
   ensure
+    # we must rewind because #each is interrupted by #find
+    # which leaves the io at the wrong position.
```

Comment:
I think maybe my comment is not functional. I think I just left it here so I could explain this problem:

Ideally `TarReader#each` would close each entry after yielding it. In fact, I wrote the code to do exactly this at first, using `Entry.open` to auto-close the entry when iteration is done.. This makes sense because as soon as each entry is out of scope, any modification to that entry can alter the position of the underlying shared IO. 

The problem is that TarReader mixes in Enumerable, and some Enumerable methods _return_ one or many of the objects in the enumerable. This would be like returning the `file`  inside a `File.open` block. The Entry _should_ be closed once the block exits. The structure of `TarReader#seek` contradicts this, since the entry lives longer than the block, so `#each` cannot close the block. The `#rewind` after seek is a fix for the io position being indeterminant after the method call.

If I could, I would remove Enumerable from TarReader and add back some of the methods in a way that allows each to close entries when the position moves beyond. TarReader is more like a stream than an enumerable.

Reply:
I think I failed to explain why exactly this method breaks this. Find `returns` from within the iteration, halting it part way through. If `#each` were to `ensure; entry.close` then `TarReader.seek` would be broken. Using `Enumerable#find` here exits the iteration, preventing the call to entry.close because it's not `ensure`d.

Reply:
tl;dr I may follow up with another PR that tries to address this issue. For now I'm ignoring it and I removed the extra comment.

Reply:
Like with the previous comment, I think you may have marked this as resolved, but forgot to push the changes actually resolving it?

Reply:
@martinemde Just double checking whether you actually planned to revert the extra comment and also somehow lost that change or whether you changed your mind and decided to keep it.

Reply:
I think I just reverted the changes to this file before committing. Thanks for catching these.